test/test_pipeline/components/data_preprocessing/test_data_preprocessing_feat_type.py

```diff
@@ -0,0 +1,128 @@
+from autosklearn.pipeline.components.data_preprocessing.feature_type import (
+    FeatTypeSplit,
+)
+
+import unittest
+
+
+class PreprocessingPipelineFeatTypeTest(unittest.TestCase):
+    def test_single_type(self):
+        DPP = FeatTypeSplit(feat_type={"A": "numerical"})
+        cs = DPP.get_hyperparameter_search_space(
+            feat_type={"A": "numerical"},
+            dataset_properties={
+                "task": 1,
+                "sparse": False,
+                "multilabel": False,
+                "multiclass": False,
+                "target_type": "classification",
+                "signed": False,
+            },
+        )
+        for key in cs.get_hyperparameters_dict().keys():
+            self.assertNotIn("text", key.split(":")[0])
+            self.assertNotIn("categorical", key.split(":")[0])
+
+        DPP = FeatTypeSplit(feat_type={"A": "categorical"})
+        cs = DPP.get_hyperparameter_search_space(
+            feat_type={"A": "categorical"},
+            dataset_properties={
+                "task": 1,
+                "sparse": False,
+                "multilabel": False,
+                "multiclass": False,
+                "target_type": "classification",
+                "signed": False,
+            },
+        )
+        for key in cs.get_hyperparameters_dict().keys():
+            self.assertNotIn("text", key.split(":")[0])
+            self.assertNotIn("numerical", key.split(":")[0])
+
+        DPP = FeatTypeSplit(feat_type={"A": "string"})
+        cs = DPP.get_hyperparameter_search_space(
+            feat_type={"A": "string"},
+            dataset_properties={
+                "task": 1,
+                "sparse": False,
+                "multilabel": False,
+                "multiclass": False,
+                "target_type": "classification",
+                "signed": False,
+            },
+        )
+        for key in cs.get_hyperparameters_dict().keys():
+            self.assertNotIn("numerical", key.split(":")[0])
+            self.assertNotIn("categorical", key.split(":")[0])
+
+    def test_dual_type(self):
+        DPP = FeatTypeSplit(feat_type={"A": "numerical", "B": "categorical"})
+        cs = DPP.get_hyperparameter_search_space(
+            feat_type={"A": "numerical", "B": "categorical"},
+            dataset_properties={
+                "task": 1,
+                "sparse": False,
+                "multilabel": False,
+                "multiclass": False,
+                "target_type": "classification",
+                "signed": False,
+            },
+        )
+        for key in cs.get_hyperparameters_dict().keys():
+            self.assertNotIn("text", key.split(":")[0])
+
+        DPP = FeatTypeSplit(feat_type={"A": "categorical", "B": "string"})
```

Comment:
You probably wanted numerical and string here, as you also have categorical and string below.

Reply:
Numerical and string also would be a good test.